src/nnet3/nnet-am-decodable-simple.cc

```diff
@@ -361,6 +361,444 @@ BaseFloat DecodableAmNnetSimpleParallel::LogLikelihood(int32 frame,
   return decodable_nnet_->GetOutput(frame, pdf_id);
 }
 
+BatchNnetComputer::BatchNnetComputer(
+    const NnetSimpleComputationOptions &opts,
+    const Nnet &nnet,
+    const VectorBase<BaseFloat> &priors,
+    int32 online_ivector_period,
+    int32 minibatch_size):
+    opts_(opts),
+    nnet_(nnet),
+    output_dim_(nnet_.OutputDim("output")),
+    log_priors_(priors),
+    online_ivector_period_(online_ivector_period),
+    compiler_(nnet_, opts_.optimize_config),
+    minibatch_size_(minibatch_size) {
+  // Check the Nnet
+  KALDI_ASSERT(IsSimpleNnet(nnet));
+  ComputeSimpleNnetContext(nnet, &nnet_left_context_, &nnet_right_context_);
+  log_priors_.ApplyLog();
+  CheckAndFixConfigs();
+  //Prepare ComputationRequest and store a NnetComputation object
+  int32 input_dim = nnet_.InputDim("input"),
+        ivector_dim = nnet_.InputDim("ivector");
+  KALDI_ASSERT(input_dim > 0);
+  
+  int32 left_context = std::max(opts_.extra_left_context,
+                                opts_.extra_left_context_initial) +
+                       nnet_left_context_,
+        right_context = std::max(opts_.extra_right_context,
+                                 opts_.extra_right_context_final) +
+                       nnet_right_context_;
+  int32 num_input_rows = left_context + opts_.frames_per_chunk + right_context,
+        num_output_rows = opts_.frames_per_chunk / opts_.frame_subsampling_factor;
```

Comment:
Is it correct num_output_rows? sometimes, we use use different frames_per_chunk to be multiple of frame_subsampling_factor.

Reply:
opts_.frames_per_chunk will be rounded up to nearest multiple of opts_.frame_subsampling_factor by function CheckAndFixConfigs().